src/sentry/relay/config.py

```diff
@@ -59,62 +340,145 @@ def _generate_pii_config(project, org_options):
+    This is intended for converting dictionaries that use the python convention to
+    dictionaries that use the javascript/JSON convention
 
+    NOTE: this function will, by default,  mutate the dictionary in place.
+    If you do not want to change the input use clone=True
 
-def get_project_key_config(project_key):
-    """Returns a dict containing the information for a specific project key"""
-    return {
-        'dsn': project_key.dsn_public,
+    :param obj: the dictionary
+
+    :return: a dictionary with the string keys converted
+
+    >>> _to_camel_case_dict({'_abc': {'_one_two_three': 1}})
+    {'abc': {'oneTwoThree': 1}}
+    >>> val = {'_abc': {'_one_two_three': 1}}
+    >>> _to_camel_case_dict({'_abc': {'_one_two_three': 1}})
+    {'abc': {'oneTwoThree': 1}}
+
+    # check that we didn't affect the original
+    >>> val
+    {'_abc': {'_one_two_three': 1}}
+
+    """
+
+    if not isinstance(obj, dict):
+        raise ValueError("Bad parameter passed expected dictionary got {}".format(repr(type(obj))))
+
+    return {_to_camel_case_name(key): _to_camel_case_dict(value) if isinstance(value, dict) else value
+            for (key, value) in six.iteritems(obj)}
+
+
+def _get_project_from_id(project_id):
+    if not project_id:
+        return None
+    if not project_id.isdigit():
+        track_outcome(0, 0, None, Outcome.INVALID, "project_id")
+        raise APIError('Invalid project_id: %r' % project_id)
+    try:
+        return Project.objects.get_from_cache(id=project_id)
+    except Project.DoesNotExist:
+        track_outcome(0, 0, None, Outcome.INVALID, "project_id")
+        raise APIError('Invalid project_id: %r' % project_id)
+
+
+def _load_filter_settings(flt, project):
+    """
+    Returns the filter settings for the specified project
+
+    :param flt: the filter function
+    :param project: the project for which we want to retrieve the options
+    :return: a dictionary with the filter options.
+        If the project does not explicitly specify the filter options then the
+        default options for the filter will be returned
+    """
+    filter_id = flt.spec.id
+    filter_key = u'filters:{}'.format(filter_id)
+    setting = ProjectOption.objects.get_value(project=project, key=filter_key, default=None)
+
+    return _filter_option_to_config_setting(flt, setting)
+
+
+def _filter_option_to_config_setting(flt, setting):
+    """
+    Encapsulates the logic for associating a filter database option with the filter setting from relay_config
+
+    :param flt: the filter
+    :param setting: the option deserialized from the database
+    :return: the option as viewed from relay_config
+    """
+    if setting is None:
+        raise ValueError("Could not find filter state for filter {0}."
+                         " You need to register default filter state in projectoptions.defaults.".format(flt.spec.id))
+
+    is_enabled = setting != '0'
+
+    ret_val = {
+        'is_enabled': is_enabled
```

Comment:
Please output the legacy browser config in a normalized fashion. IIRC, for new-style filters this could be:

```
{
  'is_enabled': True,
  'browsers': list(is_enabled),
}
```

This will help us to create a clean API.

Reply:
After reviewing the existing code I see that this is pretty much what is happening.
See https://github.com/getsentry/sentry/blob/4a832a03f5be6d1540ca305aa8193982e6bc1674/src/sentry/relay/config.py#L469-L484

So instead of 'browsers' I use a somewhat more generic entry called 'options'.